source/extensions/filters/common/fault/fault_config.h

```diff
@@ -45,33 +65,48 @@ class FaultAbortConfig {
     // Return the HTTP status code to use. Optionally passed an HTTP header that may contain the
     // HTTP status code depending on the provider implementation.
     virtual absl::optional<Http::Code> statusCode(const Http::HeaderEntry* header) const PURE;
+    // Return what percentage of requests abort faults should be applied to. Optionally passed
+    // an HTTP header that may contain the percentage depending on the provider implementation.
+    virtual envoy::type::v3::FractionalPercent
+    percentage(const Http::HeaderEntry* header) const PURE;
   };
 
   // Delay provider that uses a fixed abort status code.
   class FixedAbortProvider : public AbortProvider {
   public:
-    FixedAbortProvider(uint64_t status_code) : status_code_(status_code) {}
+    FixedAbortProvider(uint64_t status_code, const envoy::type::v3::FractionalPercent percentage)
+        : status_code_(status_code), percentage_(percentage) {}
 
     // AbortProvider
     absl::optional<Http::Code> statusCode(const Http::HeaderEntry*) const override {
       return static_cast<Http::Code>(status_code_);
     }
 
+    envoy::type::v3::FractionalPercent percentage(const Http::HeaderEntry*) const override {
+      return percentage_;
+    }
+
   private:
     const uint64_t status_code_;
+    const envoy::type::v3::FractionalPercent percentage_;
   };
 
   // Abort provider the reads a status code from an HTTP header.
-  class HeaderAbortProvider : public AbortProvider {
+  class HeaderAbortProvider : public AbortProvider, public HeaderPercentageProvider {
```

Comment:
not sure whether inheritance from multiple classes is something that we are OK with, let me know if we are not.